File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/Monitor.java
package com.sequenceiq.periscope.monitor;

import org.quartz.Job;

import com.sequenceiq.periscope.monitor.context.EvaluatorContext;

public interface Monitor<M extends Monitored> extends Job {

    String getIdentifier();

    String getTriggerExpression();

    Class<?> getEvaluatorType(M monitored);

    EvaluatorContext getContext(M monitored);
}

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/Monitor.kt
package com.sequenceiq.periscope.monitor

import org.quartz.Job

import com.sequenceiq.periscope.domain.Cluster

interface Monitor : Job {

    val identifier: String

    val triggerExpression: String

    val evaluatorType: Class<Any>

    fun getContext(cluster: Cluster): Map<String, Any>

}
